packages/graphql/lib/src/core/query_options.dart

```diff
@@ -335,3 +345,41 @@ extension WithType on Request {
   bool get isMutation => type == OperationType.mutation;
   bool get isSubscription => type == OperationType.subscription;
 }
+
+/// Handles execution of query callbacks, similarly to MutationCallbackHandler
+class QueryCallbackHandler<TParsed> {
+  final QueryOptions<TParsed> options;
+
+  QueryCallbackHandler({required this.options});
+
+  // callbacks will be called against each result in the stream,
+  // which should then rebroadcast queries with the appropriate optimism
+  Iterable<OnData<TParsed>> get callbacks =>
+      <OnData<TParsed>?>[onCompleted, onError].whereType<OnData<TParsed>>();
```

Comment:
this style is usualy good when it is easy to read, in this way you need a couple of minutes to understand what the code does, and if you do not dart friendly, you feel uncomfortable.

So I suggest
```suggestion

  Iterable<OnData<TParsed>?> get callbacks {
    var callbacks = List<OnData<TParsed>?>.empty(growable: true);
    callbacks.addAll([onCompleted, onError]);
    // FIXME: I can remove the type in whereType?
    return callbacks.whereType<OnData<TParsed>?>();
  }
```

We keep the code simple to make the package easy to onboard new people :)